### PR TITLE
Add SetBootVar() to the bootloader interface

### PR DESCRIPTION
**Pull request by mvo5**
The SetBootVar() is required to implement that all-snap boot-success
handling. It will work by setting the current snappy_os bootloader
variable to snappy_good_os (or snappy_os_good). This branch just
adds the needed interfaces for this.

### CODE REVIEW

File: partition/bootloader_uboot.go [hunk 1]
```diff
@@ -165,6 +165,10 @@ func (u *uboot) GetBootVar(name string) (value string, err error) {
 	return getBootVar(name)
 }
 
+func (u *uboot) SetBootVar(name, value string) error {
+	return setBootVar(name, value)
```
Comment (niemeyer) [inline partition/bootloader_uboot.go]: The changes in this branch itself feel fine, but the two-layered indirection that is already there (SetBootVar => the global => the actual implementation) looks a bit curious. It'd be nice to move this to become simply a struct field on uboot telling whether the legacy implementation is in use or not, and then we can have the real implementation in here, keeping just the alternative function (e.g. {get,set}BootVarLegacy) and dropping the globals.

Either way, not for this branch. Thanks for listening. :)

File: partition/partition_test.go [hunk 2]
```diff
@@ -369,6 +369,13 @@ type mockBootloader struct {
 	HandleAssetsCalled              bool
 	MarkCurrentBootSuccessfulCalled bool
 	SyncBootFilesCalled             bool
+	Bootvars                        map[string]string
```
Comment (niemeyer) [inline partition/partition_test.go]: s/var/Var/ as elsewhere?

**Comment by niemeyer** [comment]
LGTM. Just an unrelated brainstormy comment, a nitpicky suggestion, and leaving for @chipaca.

**Comment by chipaca** [comment]
The lack of tests (already for setBootVars) makes me a leeetle bit nervous.
I'm sure the feeling will pass :-p

:+1:  -- leaving unmerged for vars -> Vars.

File: partition/partition_test.go (hunk 2 above)
Comment (mvo5) [inline partition/partition_test.go]: Thanks, indeed. Fixed.

File: partition/bootloader_uboot.go (hunk 1 above)
Comment (mvo5) [inline partition/bootloader_uboot.go]: Thats a good point. I pushed a new branch feature/bootloader-set-bootvar2 that removes this double indirection.

**Comment by mvo5** [comment]
@chipaca Thanks, I really don't like you being nervous so I pushed a new feature/bootloader-set-bootvar2 branch with explicit tests for the setBootVar (it was only  indirectly tested via TestUbootMarkCurrentBootSuccessful{,FwEnv}).

**Pull request merged**